packages/cli/src/lib/implementation/only-plugins.middleware.ts

```diff
@@ -1,23 +1,25 @@
-import { CoreConfig } from '@code-pushup/models';
-import { GeneralCliOptions } from './global.model';
+import { filterItemRefsBy, toArray } from '@code-pushup/utils';
 import { OnlyPluginsOptions } from './only-plugins.model';
-import {
-  filterCategoryByPluginSlug,
-  filterPluginsBySlug,
-  validateOnlyPluginsOption,
-} from './only-plugins.utils';
+import { validateOnlyPluginsOption } from './only-plugins.utils';
 
-export function onlyPluginsMiddleware<
-  T extends GeneralCliOptions &
-    Omit<CoreConfig, 'categories'> &
-    Required<Pick<CoreConfig, 'categories'>> &
-    OnlyPluginsOptions,
->(processArgs: T): GeneralCliOptions & CoreConfig & OnlyPluginsOptions {
-  validateOnlyPluginsOption(processArgs.plugins, processArgs);
+export function onlyPluginsMiddleware<T extends OnlyPluginsOptions>(
+  processArgs: T,
+): T {
+  if (processArgs.onlyPlugins && processArgs.onlyPlugins.length > 0) {
+    const { plugins, categories = [], onlyPlugins } = processArgs;
 
-  return {
-    ...processArgs,
-    plugins: filterPluginsBySlug(processArgs.plugins, processArgs),
-    categories: filterCategoryByPluginSlug(processArgs.categories, processArgs),
-  };
+    validateOnlyPluginsOption(plugins, processArgs);
+
+    const onlyPluginsSet = new Set(toArray(onlyPlugins));
```

Comment:
Why is there `toArray`? The option type is `string[]`.

Reply:
removed